katello/backup.rb

```diff
@@ -32,6 +32,12 @@ def cleanup(exitstatus=-1)
       puts "Cleaning up backup folder and starting any stopped services... "
       FileUtils.cd("/")
       FileUtils.rm_rf @dir unless @options[:no_subdir]
+      if @options[:snapshot]
+        @databases.each do |database|
+          system("umount #{File.join(@mountdir, database)} >/dev/null")
```

Comment:
This is more or less duplicating code from https://github.com/cfouant/katello-packaging/blob/86547ae08f4eaebbddc6a5623ec847c68c42be41/katello/backup.rb#L195-L200, shall we move these lines into an own function and call it here and in L195?

Reply:
I was considering the same, however, I wanted to suppress the output here, as the output is terrible if there are no mounted filesystems. I will add that into the function.

Reply:
On second thought, I think these do need to be kept separate. We want the process in backup/destroy to go through the run_cmd method; however, as we've previously found in calling run_cmd in cleanup, we can end up caught in an endless loop.

Reply:
haha. right. I wonder if we should teach run_cmd something like "ignore_all_errors" next to the list of accepted exit codes. anyways, that is a discussion for another PR/redmine.

let me see how this code actually runs, and ack that PR afterwards.

would you mind adding a comment "not using run_cmd here because …" so that we don't forget that in the future?

Reply:
Absolutely :)